src/server/qgswmsserver.cpp

```diff
@@ -2079,6 +2079,16 @@ int QgsWMSServer::configureMapRender( const QPaintDevice* paintDevice ) const
     mMapRenderer->setOutputUnits( QgsMapRenderer::Pixels ); //SLD units are in pixels normally
   }
 
+  //Clear expression context and set project variables
+  QgsRenderContext* ctx = mMapRenderer->rendererContext();
+  if ( ctx )
+  {
+    QgsExpressionContext newContext;
+    ctx->setExpressionContext( newContext );
+    QgsExpressionContext&  expCtx = ctx->expressionContext();
+    expCtx.appendScope( QgsExpressionContextUtils::projectScope() );
```

Comment:
No particular reason. I was first hoping to get the context and only replace the project scope. But yes, since the context has to be created anyway, it can be set there directly.